src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -0,0 +1,306 @@
+        if (busStopJusticeCenterTrip != null) busStopJusticeCenterTrip.delete();
+    }
+
+    @AfterEach
+    public void tearDownAfterTest() {
+        if (trackedJourney != null) {
+            trackedJourney.delete();
+            trackedJourney = null;
+        }
+    }
+
+    // WIP. Time adjusted so that the provided traveler times start around the same time at the trip.
+    @Test
+    void canTrackSubwayTrip() throws IOException {
+        String traceTimes = CommonTestUtils.getTestResourceAsString("controllers/api/monitored-trip-trace-times.csv");
+        String[] traces = traceTimes.split("\n");
+        TrackedJourney trackedJourney = new TrackedJourney();
+        trackedJourney.tripId = monitoredTrip.id;
+        HashMap<String, Integer> outcome = new HashMap<>();
+        for (String trace : traces) {
+            String[] traceValues = trace.split(",");
+            TrackingLocation trackingLocation = new TrackingLocation();
+            ZonedDateTime zdt = ZonedDateTime.parse(traceValues[0]);
+            ZonedDateTime timeAdjusted = zdt.plusMinutes(3).plusSeconds(8);
+            trackingLocation.timestamp = new Date(timeAdjusted.toInstant().toEpochMilli());
+            trackingLocation.lat = Double.parseDouble(traceValues[1]);
+            trackingLocation.lon = Double.parseDouble(traceValues[2]);
+            trackedJourney.locations = List.of(trackingLocation);
+            TripStatus tripStatus = ManageTripTracking.getTripStatus(trackedJourney, monitoredTrip);
+            if (outcome.containsKey(tripStatus.name())) {
+                Integer hits = outcome.get(tripStatus.name());
+                hits++;
+                outcome.put(tripStatus.name(), hits);
+            } else {
+                outcome.put(tripStatus.name(), 1);
+            }
+        }
+        for (TripStatus tripStatus : TripStatus.values()) {
+            System.out.println(tripStatus.name() + ": " + outcome.get(tripStatus.name()));
+        }
+        System.out.println("Total: " + traces.length);
+    }
+
+
+    // WIP.
+    @Test
+    void canTrackBusStopToJusticeCenterTrip() throws IOException {
+        String traceTimes = CommonTestUtils.getTestResourceAsString("controllers/api/bus-stop-justice-center-trace-times.csv");
+        String[] traces = traceTimes.split("\n");
+        TrackedJourney trackedJourney = new TrackedJourney();
+        trackedJourney.tripId = busStopJusticeCenterTrip.id;
+        HashMap<String, Integer> outcome = new HashMap<>();
+        for (String trace : traces) {
+            String[] traceValues = trace.split(",");
+            TrackingLocation trackingLocation = new TrackingLocation();
+            ZonedDateTime zdt = ZonedDateTime.parse(traceValues[0]);
+            trackingLocation.timestamp = new Date(zdt.toInstant().toEpochMilli());
+            trackingLocation.lat = Double.parseDouble(traceValues[1]);
+            trackingLocation.lon = Double.parseDouble(traceValues[2]);
```

Comment:
This is an important piece of logic that converts converts coordinates into a `TrackingLocation`, extract it into a  constructor for `TrackingLocation` or into a helper method in the `main` code.